catalyst_voices/packages/internal/catalyst_voices_blocs/lib/src/session/session_cubit.dart

```diff
@@ -127,32 +133,60 @@ final class SessionCubit extends Cubit<SessionState>
     _updateState();
   }
 
+  void _onAdminToolsChanged(AdminToolsState adminTools) {
+    _logger.fine('Admin tools changed: $adminTools');
+
+    _adminToolsState = adminTools;
+    _updateState();
+  }
+
   void _updateState() {
+    if (_adminToolsState.enabled) {
+      emit(_createMockedSessionState());
+    } else {
+      emit(_createSessionState());
+    }
```

Comment:
Can `adminTools` somehow be moved out of `SessionCubit` to any of `Services`?

Reply:
Admin tools mock the UI, services don't have access to view models.

I tried to approach it differently, instead of mocking the UI to a certain session state then simply login the user with a dummy account so that services are "aware" but that basically means the operations user does would be done as "dummy user", not as admin user. Some things you won't be able to do as "dummy user", you must do them as admin (i.e. admin should have access to way more data than "dummy user").

The difference is we want to mock UI, not mock the logic.